org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/navigation/XbaseHyperLinkHelper.java

```diff
@@ -153,6 +169,40 @@ public void createHyperlinksByOffset(final XtextResource resource, final int off
 				}
 			}
 		}
+		
+		createHyperlinksInJavaDoc(resource, offset, acceptor);
+	}
+
+	/**
+	 * Creates hyperlinks on types referenced in javadoc comments.
+	 * 
+	 * @since 2.16
+	 */
+	protected void createHyperlinksInJavaDoc(XtextResource resource, int offset, IHyperlinkAcceptor acceptor) {
+		IParseResult parseResult = resource.getParseResult();
+		if(parseResult != null) {
+			INode rootNode = parseResult.getRootNode();
+			ILeafNode node = NodeModelUtils.findLeafNodeAtOffset(rootNode, offset);
```

Comment:
@miklossy Will this never return null?

Reply:
@tivervac If the node variable is null, it will be properly handled by the subsequent [NodeModelUtils.findActualSemanticObjectFor(node)](https://github.com/eclipse/xtext-core/blob/796e2e8aef5a6390a9a008594517377c1a6c0399/org.eclipse.xtext/src/org/eclipse/xtext/nodemodel/util/NodeModelUtils.java#L263-L264) call.